proxy/proxy.go

```diff
@@ -495,3 +508,14 @@ func (proxy *Proxy) getDNSDomain() (domain string) {
 
 	return string(b)
 }
+
+func (proxy *Proxy) networkModeIsAttachable(networkMode string) (bool, error) {
+	for strings.HasPrefix(networkMode, "container:") {
+		container, err := proxy.client.InspectContainer(strings.TrimPrefix(networkMode, "container:"))
+		if err != nil || container.HostConfig == nil {
+			return false, err
+		}
+		networkMode = container.HostConfig.NetworkMode
+	}
+	return networkMode != "host", nil
+}
```

Comment:
A restart policy won't change the networkMode, though. Only an api stop/start with the hostconfig provided.

We could do these checks every time a container is started, and send a hostConfig if needed (regardless of whether the user sent one).